isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -3438,36 +3476,1772 @@ namespace isobus
+					{
+						retVal = FontSize::Size64x96;
+					}
+					else if (scaleFactor >= 0.6f)
+					{
+						retVal = FontSize::Size64x64;
+					}
+					else if (scaleFactor >= 0.5f)
+					{
+						retVal = FontSize::Size48x64;
+					}
+					else if (scaleFactor >= 0.37f)
+					{
+						retVal = FontSize::Size32x48;
+					}
+					else if (scaleFactor >= 0.33f)
+					{
+						retVal = FontSize::Size32x32;
+					}
+					else if (scaleFactor >= 0.25f)
+					{
+						retVal = FontSize::Size24x32;
+					}
+					else if (scaleFactor >= 0.18f)
+					{
+						retVal = FontSize::Size16x24;
+					}
+					else if (scaleFactor >= 0.16f)
+					{
+						retVal = FontSize::Size16x16;
+					}
+					else if (scaleFactor >= 0.125f)
+					{
+						retVal = FontSize::Size12x16;
+					}
+					else if (scaleFactor >= 0.09f)
+					{
+						retVal = FontSize::Size8x12;
+					}
+					else if (scaleFactor >= 0.08f)
+					{
+						retVal = FontSize::Size8x8;
+					}
+					else
+					{
+						retVal = FontSize::Size6x8;
+					}
+				}
+			}
+			break;
+
+			case FontSize::Size128x128:
+			{
+				if (scaleFactor >= SCALE_FACTOR_POSITIVE_FUDGE)
+				{
+					if (scaleFactor >= 1.45f)
+					{
+						retVal = FontSize::Size128x192;
+					}
+				}
+				else if (scaleFactor <= SCALE_FACTOR_NEGATIVE_FUDGE)
+				{
+					if (scaleFactor >= 0.75f)
+					{
+						retVal = FontSize::Size64x96;
+					}
+					else if (scaleFactor >= 0.5f)
+					{
+						retVal = FontSize::Size64x64;
+					}
+					else if (scaleFactor >= 0.37f)
+					{
+						retVal = FontSize::Size32x48;
+					}
+					else if (scaleFactor >= 0.25f)
+					{
+						retVal = FontSize::Size32x32;
+					}
+					else if (scaleFactor >= 0.18f)
+					{
+						retVal = FontSize::Size16x24;
+					}
+					else if (scaleFactor >= 0.125f)
+					{
+						retVal = FontSize::Size16x16;
+					}
+					else if (scaleFactor >= 0.09f)
+					{
+						retVal = FontSize::Size8x12;
+					}
+					else if (scaleFactor >= 0.06f)
+					{
+						retVal = FontSize::Size8x8;
+					}
+					else
+					{
+						retVal = FontSize::Size6x8;
+					}
+				}
+			}
+			break;
+
+			case FontSize::Size128x192:
+			{
+				if (scaleFactor <= SCALE_FACTOR_NEGATIVE_FUDGE)
+				{
+					if (scaleFactor >= 0.75f)
+					{
+						retVal = FontSize::Size96x128;
+					}
+					else if (scaleFactor >= 0.5f)
+					{
+						retVal = FontSize::Size64x96;
+					}
+					else if (scaleFactor >= 0.37f)
+					{
+						retVal = FontSize::Size48x64;
+					}
+					else if (scaleFactor >= 0.25f)
+					{
+						retVal = FontSize::Size32x48;
+					}
+					else if (scaleFactor >= 0.18f)
+					{
+						retVal = FontSize::Size24x32;
+					}
+					else if (scaleFactor >= 0.125f)
+					{
+						retVal = FontSize::Size16x24;
+					}
+					else if (scaleFactor >= 0.09f)
+					{
+						retVal = FontSize::Size12x16;
+					}
+					else if (scaleFactor >= 0.06f)
+					{
+						retVal = FontSize::Size8x12;
+					}
+					else
+					{
+						retVal = FontSize::Size6x8;
+					}
+				}
+			}
+			break;
```

Comment:
For maintainability reasons, wouldn't this be easier to parse it using a map instead?

Reply:
How do you propose to do that? I agree these trees of conditionals is not very nice to read or maintain, so if we can switch it to a nicer structure that'd be ideal. But if the scale factor can be totally arbitrary, like 0.6f or something which is not in this list of conditionals, how would we clamp that to match some key value in the map?

Reply:
See #173 for my proposal